src/WebJobs.Script.WebHost/DependencyInjection/ScopedResolver.cs

```diff
@@ -32,7 +34,7 @@ public void Dispose()
                 .ContinueWith(t =>
                 {
                     Container.Dispose();
-                });
+                }, TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
This addresses potential issues with disposal timing. Added a test to highlight the problem and ensure we don't regress.

Reply:
Which test covers this?

Reply:
Discussed offline.